harness/determined/pytorch/_callback.py

```diff
@@ -16,28 +16,32 @@ class PyTorchCallback:
 
     .. warning::
         If distributed training is enabled, every GPU will execute a copy of this callback
-        (except for :meth:`on_validation_step_end` and :meth:`on_checkpoint_end`).  To
+        (except for :meth:`on_validation_end` and :meth:`on_checkpoint_end`).  To
         configure a callback implementation to execute on a subset of GPUs, please condition
         your implementation on ``trial.context.distributed.get_rank()``.
     """
 
-    def on_train_step_start(self, step_id: int) -> None:
+    def on_batch_start(self, batch_idx: int) -> None:
```

Comment:
This is really different behavior than what we had before, and it doesn't seem like it would  be terribly useful to me, since we already call train_batch on every batch.

I'm not sure what use cases we were trying to solve before though?  Are those use cases still valid?

Git blame says @yoavz wrote these hooks, maybe @shiyuann or @aaron276h know the answer though?

And if those use cases are still valid, how would we address them after removing steps from the UX?

Reply:
This was done so that users can do adjustments to the optimizer and model before training. Don't remember the exact use cases. I do think it makes sense to leave this callback in.

Reply:
I chose these changes because it's exactly what is in the ERD. I assumed they were discussed/decided already. I understand `on_batch_start`/`on_batch_end` seem pretty not useful though. I think most of the context on the original decision is somewhere in #ml-ag slack.